ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/visitor/core/services/collectors/classes/JavaClassStructureCollector.java

```diff
@@ -0,0 +1,45 @@
+package edu.usb.argos.ASTProcessor.visitor.core.services.collectors.classes;
+
+import edu.usb.argos.ASTProcessor.antlr.JavaParser;
+import edu.usb.argos.ASTProcessor.visitor.core.interfaces.collectors.classes.IClassStructureCollector;
+import edu.usb.argos.ASTProcessor.visitor.shared.validation.ContextValidator;
+import org.antlr.v4.runtime.ParserRuleContext;
+
+import java.util.*;
+
+public class JavaClassStructureCollector implements IClassStructureCollector<ParserRuleContext> {
+
+    @Override
+    public String getSuperClass(ParserRuleContext ctx) {
```

Comment:
What I understand from this method is that it returns null if the class is not a subclass of another class? If so, you can use the Option object as before.

Reply:
Done